src/groups/posts.ts

```diff
@@ -0,0 +1,76 @@
+import db from '../database';
+import privileges from '../privileges';
+import groups from '.';
+import posts from '../posts';
+import topics from '../topics';
+import { PostObject } from '../types/post';
+import { GroupDataObject, TopicObject } from '../types';
+
+interface GroupsInterface {
+  onNewPostMade: (postData: { uid: number, cid: number, pid: number, timestamp: string }) => Promise<void>;
+  getLatestMemberPosts: (groupName: string, max: number, uid: string) => Promise<PostObject[]>;
+  getUserGroupMembership: (set: string, uids: number[]) => Promise<string[][]>;
+}
+
+export = function (Groups: GroupsInterface) {
+    async function truncateMemberPosts(groupName: string) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+        const pids = await db.getSortedSetRevRange(`group:${groupName}:member:pids`, 10, 10) as number[];
+        const lastPid = pids[0];
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+        const score = await db.sortedSetScore(`group:${groupName}:member:pids`, lastPid) as number;
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+        await db.sortedSetsRemoveRangeByScore([`group:${groupName}:member:pids`], '-inf', score);
+    }
+
+    Groups.onNewPostMade = async function (postData) {
+        const allgroupNames = await Groups.getUserGroupMembership('groups:visible:createtime', [postData.uid]);
+        let groupNames = allgroupNames[0];
```

Comment:
Same here, does TypeScript do implicit type casting for these variables?

Reply:
Same here